homeassistant/components/otbr/config_flow.py

```diff
@@ -140,34 +186,35 @@ async def async_step_hassio(
         url = f"http://{config['host']}:{config['port']}"
         config_entry_data = {"url": url}
 
-        if self._async_in_progress(include_uninitialized=True):
-            # We currently don't handle multiple config entries, abort if hassio
-            # discovers multiple addons with otbr support
-            return self.async_abort(reason="single_instance_allowed")
-
         if current_entries := self._async_current_entries():
             for current_entry in current_entries:
                 if current_entry.source != SOURCE_HASSIO:
                     continue
                 current_url = yarl.URL(current_entry.data["url"])
-                if (
+                if not (unique_id := current_entry.unique_id):
                     # The first version did not set a unique_id
                     # so if the entry does not have a unique_id
                     # we have to assume it's the first version
-                    current_entry.unique_id
-                    and (current_entry.unique_id != discovery_info.uuid)
+                    # This check can be removed in HA Core 2025.9
+                    unique_id = discovery_info.uuid
```

Comment:
From where does the config entry migration get the needed data?

Reply:
Discussed on Discord can be done in a follow up

Reply:
We discussed this again on Discord.

To be able to do the migration, the add-on must be running so we can get the UUID from it. We should not stall the migration waiting for the add-on to start.
The migrator can't raise `ConfigEntryNotReady`, and it doesn't seem worth the effort to add support for that just because of `otbr`.